### PR TITLE
change: Bot refactoring

### PR DESCRIPTION
**Pull request by Kondrahin**
# Description

Bot refactoring

# Checklist

- [x] Linters and tests have passed
- [ ] I've written a changelog for PR change
- [ ] I'll make a release when PR is approved

### CODE REVIEW

File: botx/bot/bot.py [hunk 1]
```diff
@@ -207,10 +203,7 @@ def async_execute_bot_command(self, bot_command: BotCommand) -> None:
         # raise UnknownBotAccountError if no bot account with this bot_id.
         self._bot_accounts_storage.ensure_bot_id_exists(bot_command.bot.id)
 
-        task = asyncio.create_task(
-            self._handler_collector.handle_bot_command(bot_command, self),
-        )
-        self._tasks.add(task)
+        self._handler_collector.handle_bot_command_in_background(self, bot_command)
```
Comment (lxmnk) [inline botx/bot/bot.py:206]: I suggest to use `async_handle_bot_command` method name.

File: botx/bot/handler_collector.py [hunk 2]
```diff
@@ -44,13 +50,38 @@ def __init__(self, middlewares: Optional[Sequence[Middleware]] = None) -> None:
...
         self._middlewares = self._reversed_middlewares(middlewares)
+        self._tasks: "WeakSet[asyncio.Task[None]]" = WeakSet()
 
     def include(self, *others: "HandlerCollector") -> None:
         """Include other `HandlerCollector`."""
 
         for collector in others:
             self._include_collector(collector)
 
+    def insert_exception_middleware(
```
Comment (lxmnk) [inline botx/bot/handler_collector.py:61]: This method should be private.

Comment (lxmnk) [inline botx/bot/handler_collector.py:61]: Pls place this method before private ones. It used only by bot, no library users.

File: botx/bot/handler_collector.py [hunk 3]
```diff
@@ -44,13 +50,38 @@ def __init__(self, middlewares: Optional[Sequence[Middleware]] = None) -> None:
...
+        self,
+        bot: "Bot",
+        bot_command: BotCommand,
+    ) -> None:
+        task = asyncio.create_task(
+            self.handle_bot_command(bot_command, bot),
+        )
+        self._tasks.add(task)
+
+    async def wait_active_tasks(self) -> None:
```
Comment (lxmnk) [inline botx/bot/handler_collector.py:78]: Also move this method below.

**Review by Nimond — APPROVED** [review]

**Review by lxmnk — APPROVED** [review]

**Pull request merged**